tools/fog-local-network/fog_local_network.py

```diff
@@ -135,16 +135,16 @@ def start(self):
         result = subprocess.check_output(cmd, shell=True)
 
     def stop(self):
-        if hasattr(self, "fog_ledger"):
+        if hasattr(self, "fog_ledger") and self.fog_ledger is not None:
             self.fog_ledger.stop()
 
-        if hasattr(self, "fog_report"):
+        if hasattr(self, "fog_report") and self.fog_report is not None:
             self.fog_report.stop()
 
-        if hasattr(self, "fog_view"):
+        if hasattr(self, "fog_view") and self.fog_view is not None:
             self.fog_view.stop()
 
-        if hasattr(self, "fog_ingest"):
+        if hasattr(self, "fog_ingest") and self.fog_ingest is not None:
             self.fog_ingest.stop()
```

Comment:
Not sure this is better, but anytime I see `hasattr()` followed by accessing the attribute it feels like a code smell
and then github thinks the lines are deleted so no suggestion is possible :(, feel free to ignore
```python
        if ledger := getattr(self, "fog_ledger", None):
            ledger.stop()

        if report := getattr(self, "fog_report", None):
            report.stop()

        if view := getattr(self, "fog_view", None):
            view.stop()

        if ingest := getattr(self, "fog_ingest", None):
            ingest.stop()
```